### PR TITLE
added blocks option for lstm2d

### PR DESCRIPTION
**Pull request by johnsmithm**

### CODE REVIEW

File: tensorflow/contrib/ndlstm/python/lstm2d.py [hunk 1]
```diff
@@ -91,21 +91,71 @@ def horizontal_lstm(images, num_filters_out, scope=None):
...
+      of the pooling. Can be an int if both values are the same.
+
+  Returns:
+    (num_images, height/kernel_height, width/kernel_width,
+    depth*kernel_height*kernel_width) tensor
+  """
+  with variable_scope.variable_scope("image_blocks"):
+    batch_size, height, width, chanels = _shape(images)
+
+    if height%kernel_size[0] != 0:
```
Comment (lukaszkaiser) [inline tensorflow/contrib/ndlstm/python/lstm2d.py]: Could you add a space before and after "%"?

File: tensorflow/contrib/ndlstm/python/lstm2d.py [hunk 2]
```diff
@@ -91,21 +91,71 @@ def horizontal_lstm(images, num_filters_out, scope=None):
...
+      batch_size, height, width, chanels = _shape(images)
+    if width%kernel_size[1] != 0:
+      offset = array_ops.zeros([batch_size,
+                                height,
+                                kernel_size[1]-(width%kernel_size[1]),
+                                chanels])
+      images = array_ops.concat([images, offset], 2)
+      batch_size, height, width, chanels = _shape(images)
+
+    h, w = int(height/kernel_size[0]), int(width/kernel_size[1])
```
Comment (lukaszkaiser) [inline tensorflow/contrib/ndlstm/python/lstm2d.py]: Same for "/", or maybe just use "//"?

File: tensorflow/contrib/ndlstm/python/lstm2d.py [hunk 3]
```diff
@@ -91,21 +91,71 @@ def horizontal_lstm(images, num_filters_out, scope=None):
...
+    batch_size, height, width, chanels = _shape(images)
+
+    if height%kernel_size[0] != 0:
+      offset = array_ops.zeros([batch_size,
+                                kernel_size[0]-(height%kernel_size[0]),
+                                width,
+                                chanels])
+      images = array_ops.concat([images, offset], 1)
+      batch_size, height, width, chanels = _shape(images)
+    if width%kernel_size[1] != 0:
```
Comment (lukaszkaiser) [inline tensorflow/contrib/ndlstm/python/lstm2d.py]: Here too.

File: tensorflow/contrib/ndlstm/python/lstm2d.py [hunk 4]
```diff
@@ -91,21 +91,71 @@ def horizontal_lstm(images, num_filters_out, scope=None):
...
+      offset = array_ops.zeros([batch_size,
+                                height,
+                                kernel_size[1]-(width%kernel_size[1]),
```
Comment (lukaszkaiser) [inline tensorflow/contrib/ndlstm/python/lstm2d.py]: And here, for "-" too.

**Review by lukaszkaiser — CHANGES_REQUESTED** [review]
Just a few style nits (space around operators). All looks fine otherwise, thanks!

**Comment by johnsmithm** [comment]
thanks for review, maybe someone should add these rule in pylint too

**Comment by johnsmithm** [comment]
Waiting for review!

**Review by lukaszkaiser — APPROVED** [review]
Looks good, sorry for the delay!

**Comment by drpngx** [comment]
Jenkins, test this please.

**Pull request merged**